src/lib/containers/QueryWrapper.tsx

```diff
@@ -152,6 +154,29 @@ export default class QueryWrapper extends React.Component<QueryWrapperProps, Que
                 console.log("Failed to get data ", err)
             })
     }
+
+    /**
+     * Exectue the given query
+     *
+     * @param {*} queryRequest Query request as specified by https://docs.synapse.org/rest/org/sagebionetworks/repo/model/table/Query.html
+     * @memberof QueryWrapper
+     */
+    getNextPageOfData(queryRequest: any) {
+        this.setState({
+            isLoading: true
+        })
+        SynapseClient.getQueryTableResults(queryRequest, this.props.token)
+            .then(
+                (data: QueryResultBundle )=> {
+                    let oldData: QueryResultBundle = cloneDeep(this.state.data)
```

Comment:
the name 'oldData' initially threw me off, since it really contains the new data query result rows, but I don't have a better suggestion (since it is initially a copy of the old state data)...